packages/client/src/components/uppy/uppy.ts

```diff
@@ -0,0 +1,257 @@
+    uploadedParts.push(part)
+  })
+  ;(window as any)['uppy'] = uppy
+  return uppy
+
+  // Initiate multipart upload
+  async function createMultipartUpload({
+    name,
+  }: UppyFile): Promise<{ key: string; uploadId: string }> {
+    const request = createRequest(CreateMultipartUploadDocument, {
+      fileKey: name,
+    })
+    return await new Promise((resolve) => {
+      try {
+        urqlClient.executeMutation(request)((result) => {
+          if (typeof result === 'object') {
+            const { data } = result[0] as any
+            if (data) {
+              const { key, uploadId } =
+                data.createMultipartUpload as CreateMultipartUploadReturn
+              resolve({ key, uploadId })
+            }
+          }
+        })
+      } catch (error: any) {
+        console.log(error.message)
+      }
+    })
+  }
+  // Get presigned url for each part
+  async function prepareUploadParts(data: any, metadata: any) {
+    const urls: { [key: string]: string } = {}
+    try {
+      for (const partNumber of metadata.partNumbers) {
+        const request = createRequest(PrepareUploadPartsDocument, {
+          fileKey: metadata.key,
+          uploadId: metadata.uploadId,
+          partNumber: partNumber,
+        } as PrepareUploadPartsQueryVariables)
+        await new Promise((r) => {
+          urqlClient.executeQuery(request)((result) => {
+            if (typeof result === 'object') {
+              const { data } = result[0] as any
+              if (data) {
+                const { url } = data.prepareUploadParts as GetUrlReturn
+                urls[partNumber] = url
+                r(undefined)
+              }
+            }
+          })
+        })
+      }
+    } catch (error: any) {
+      console.log(error.message)
+    }
+    return { presignedUrls: urls }
+  }
+  async function listParts(
+    file: any,
+    { uploadId, key }: { uploadId: string; key: string }
+  ): Promise<AwsS3Part[]> {
+    const request = createRequest(ListPartsDocument, {
+      fileKey: key,
+      uploadId,
+    } as ListPartsQueryVariables)
+    return await new Promise((r) => {
+      try {
+        urqlClient.executeQuery(request)((result) => {
+          if (typeof result === 'object') {
+            const { data } = result[0] as any
+            if (data) {
+              const parts = data.listParts as AwsS3Part[]
+              const _parts = parts
+                .map((part) => ({
+                  ETag: part.ETag,
+                  PartNumber: part.PartNumber,
+                  Size: part.Size,
+                }))
+                .filter((part) =>
+                  getUploadedParts(uploadId).find(
+                    (p) => p.PartNumber === part.PartNumber
+                  )
+                )
+              r(_parts)
+            }
+          }
+        })
+      } catch (error: any) {
+        console.log(error.message)
+      }
+    })
+  }
+  //
+  async function completeMultipartUpload(
+    file: any,
```

Comment:
please, use specific types instead of all any types (maybe typescript interfaces with read-only fields)